src/esp/gfx/CubeMap.h

```diff
@@ -0,0 +1,203 @@
+
+  /**
+   * @brief, reset the image size
+   * @return true, if image size has been changed, and reset has happened,
+   * otherwise false
+   */
+  bool reset(int imageSize);
+
+  /**
+   * @brief Get the cubemap texture based on the texture type
+   * @return Reference to the cubemap texture
+   */
+  Magnum::GL::CubeMapTexture& getTexture(TextureType type);
+
+#ifndef MAGNUM_TARGET_WEBGL
+  /**
+   * ```
+   *           +----+
+   *           | -Y |
+   * +----+----+----+----+
+   * | -Z | -X | +Z | +X |
+   * +----+----+----+----+
+   *           | +Y |
+   *           +----+
+   * ```
+   * NOTE: +Y is top
+   * @brief save the cubemap texture based on the texture type
+   * @param type, texture type
+   * @param imageFilePrefix, the filename prefix
+   * The 6 image files then would be:
+   * {imageFilePrefix}.{texType}.+X.png
+   * {imageFilePrefix}.{texType}.-X.png
+   * {imageFilePrefix}.{texType}.+Y.png
+   * {imageFilePrefix}.{texType}.-Y.png
+   * {imageFilePrefix}.{texType}.+Z.png
+   * {imageFilePrefix}.{texType}.-Z.png
+   * @return true, if success, otherwise false
+   */
+  // TODO: color HDR textures
+  bool saveTexture(TextureType type, const std::string& imageFilePrefix);
+#endif
+
+  /**
+   * ```
+   *           +----+
+   *           | -Y |
+   * +----+----+----+----+
+   * | -Z | -X | +Z | +X |
+   * +----+----+----+----+
+   *           | +Y |
+   *           +----+
+   * ```
+   * NOTE: +Y is top
+   * @brief load cubemap texture from external images
+   * @param type can be "rgba", "depth", or "objectId" (TODO)
+   * @param imageFilePrefix, the prefix of the image filename
+   * @param imageFileExtension, the image filename extension (such as "png",
+   * "jpg")
+   * @return true if succeeded, otherwise false
+   * The 6 image files then would be:
+   * {imageFilePrefix}.{texType}.+X.{imageFileExtension}
+   * {imageFilePrefix}.{texType}.-X.{imageFileExtension}
+   * {imageFilePrefix}.{texType}.+Y.{imageFileExtension}
+   * {imageFilePrefix}.{texType}.-Y.{imageFileExtension}
+   * {imageFilePrefix}.{texType}.+Z.{imageFileExtension}
+   * {imageFilePrefix}.{texType}.-Z.{imageFileExtension}
+   */
+  void loadTexture(TextureType type,
+                   const std::string& imageFilePrefix,
+                   const std::string& imageFileExtension);
+
+  /**
+   * @brief Render to cubemap texture using the camera
+   * @param camera, a cubemap camera
+   * NOTE: It will NOT automatically generate the mipmap for the user
+   */
+  void renderToTexture(CubeMapCamera& camera,
+                       scene::SceneGraph& sceneGraph,
+                       RenderCamera::Flags flags);
+
+ private:
+  Flags flags_;
+  int imageSize_ = 0;
+  std::map<TextureType, std::unique_ptr<Magnum::GL::CubeMapTexture>> textures_;
+
+  /**
+   * @brief Recreate textures
+   */
+  void recreateTexture();
+
+  // framebuffers (one for every cube side)
+  Magnum::GL::Framebuffer frameBuffer_[6]{
```

Comment:
Is there a better way to initialize this?

Reply:
Probably not... except if you'd want to use [StaticArray](https://doc.magnum.graphics/corrade/classCorrade_1_1Containers_1_1StaticArray.html#Containers-StaticArray-initialization), which has the magic (I hope it works with move-only types):

```cpp
Corrade::Containers::StaticArray<6, Magnum::GL:Framebuffer> frameBuffer_{Corrade::Containers::DirectInit, Magnum::NoCreate};
```